railties/lib/rails/engine.rb

```diff
@@ -548,7 +548,18 @@ def config
     # Blog::Engine.load_seed
     def load_seed
       seed_file = paths["db/seeds.rb"].existent.first
-      load(seed_file) if seed_file
+      return unless seed_file
+      ActiveRecord::Base.transaction do
```

Comment:
railties don't depend on Active Record so we can't call methods of Active Record here. I think a different way to fix this problem is to generate the transaction in the seed file. This way the user chan chose which database connection they want to use and we don't make railtie depend on Active Record. Or move to database tasks as recommended by Kasper